packages/table/test/table.test.ts

```diff
@@ -0,0 +1,1043 @@
+                        <sp-table-cell>Row Item Echo</sp-table-cell>
+                        <sp-table-cell>Row Item Echo</sp-table-cell>
+                        <sp-table-cell>Row Item Echo</sp-table-cell>
+                    </sp-table-row>
+                </sp-table-body>
+            </sp-table>
+        `);
+        const tableHeadCheckboxCell = el.querySelector(
+            'sp-table-head sp-table-checkbox-cell'
+        ) as TableCheckboxCell;
+
+        expect(el.selected).to.deep.equal(['row1', 'row2']);
+
+        tableHeadCheckboxCell.checkbox.click();
+
+        expect(changeSpy.calledOnce).to.be.true;
+        expect(changeSpy.calledWithExactly(el)).to.be.true;
+
+        expect(el.selected).to.deep.equal([
+            'row1',
+            'row2',
+            'row3',
+            'row4',
+            'row5',
+        ]);
+    });
+
+    xit('dispatches `rangeChanged` events on Virtualized Table', async () => {
+        // console.log('This test');
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        // console.log(1);
+        await nextFrame();
+        // console.log(2);
+        await nextFrame();
+        // console.log(3);
+        await nextFrame();
+        // debugger;
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+
+        // console.log('4a');
+        const rangeChanged = oneEvent(el, 'rangeChanged');
+        // console.log(4);
+        let tableRow = el.querySelector('sp-table-row') as TableRow;
+        const initialValue = tableRow.value;
+        el.scrollToIndex(47);
+        // (el as unknown as { tableBody: TableBody }).tableBody.scrollTop = (el as unknown as { tableBody: TableBody }).tableBody.scrollHeight;
+        // console.log('we await after this...');
+        await nextFrame();
+        await nextFrame();
+        // debugger;
+        await rangeChanged;
+        tableRow = el.querySelector('sp-table-row') as TableRow;
+        const newValue = tableRow.value;
+
+        expect(newValue).to.not.equal(initialValue);
+    });
+
+    it('dispatches `visibilityChanged` events on Virtualized Table', async () => {
+        // console.log('NOT this test');
+        const visibilityChangedSpy = spy();
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
```

Comment:
Possible to adopt this from the stories? I'd also listen to your argument for not doing that, but I can't remember if we actually got into that or not.

Reply:
A middle ground could be to make a helper in this file that created fixtures without them needing to be in each test. There's a couple of nice patterns for that. Using `beforeEach` [like this](https://github.com/adobe/spectrum-web-components/blob/main/packages/number-field/test/number-field.test.ts#L107-L112) can be a nice way to make a suite of tests be closely related and share their baseline setup making it easier to isolate single test cases against a single configuration of the elements.